iothub_client/src/iothub_client_core_ll.c

```diff
@@ -2319,6 +2340,24 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_LL_SetOption(IOTHUB_CLIENT_CORE_LL_HANDLE
                 result = IOTHUB_CLIENT_OK;
             }
         }
+        else if (strcmp(optionName, OPTION_DT_MODEL_ID) == 0)
+        {
+            if (handleData->dt_model_id != NULL)
+            {
+                STRING_delete(handleData->dt_model_id);
+                handleData->dt_model_id = NULL;
+            }
+
+            if ((handleData->dt_model_id = STRING_construct_sprintf("&%s=%s", DT_MODEL_ID_TOKEN, STRING_c_str(URL_EncodeString((const char*)value)))) == NULL)
+            {
```

Comment:
This is hard-coding an MQTT construct into the client layer, which is theoretically transport agnostic.  It seems here we should save off the dt_model_id as-is and then let the underlying layer do the needed work.